db/Pos/DB/Functions.hs

```diff
@@ -18,28 +20,47 @@ module Pos.DB.Functions
 
 import           Universum
 
-import qualified Data.ByteString  as BS (drop, isPrefixOf)
-import           Formatting       (bprint, builder, sformat, shown, stext, string, (%))
+import qualified Data.ByteString    as BS (drop, isPrefixOf)
+import           Formatting         (bprint, builder, int, sformat, shown, stext, string,
+                                     (%))
 
-import           Pos.Binary.Class (Bi, decodeFull, serialize')
-import           Pos.DB.Class     (DBIteratorClass (..), DBTag, IterType, MonadDB (..),
-                                   MonadDBRead (..))
-import           Pos.DB.Error     (DBError (DBMalformed))
-import           Pos.Util.Util    (maybeThrow)
+import           Pos.Binary.Class   (Bi, decodeFull, serialize')
+import           Pos.Core.Constants (dbSerializeVersion)
+import           Pos.DB.Class       (DBIteratorClass (..), DBTag, IterType, MonadDB (..),
+                                     MonadDBRead (..))
+import           Pos.DB.Error       (DBError (..))
+import           Pos.Util.Util      (maybeThrow)
 
 
 -- | Read serialized value associated with given key from pure DB.
-dbGetBi
+dbGetBiNoVersion
     :: forall v m.
        (Bi v, MonadDBRead m)
     => DBTag -> ByteString -> m (Maybe v)
+dbGetBiNoVersion tag key = do
+    bytes <- dbGet tag key
+    traverse (fmap snd . dbDecode . (ToDecodeValue key)) bytes
+
+-- | Write serializable value to DB for given key.
+dbPutBiNoVersion :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> m ()
+dbPutBiNoVersion tag k v = dbPut tag k (serialize' v)
+
+-- | Read serialized value (with version) associated with given key from pure DB.
+dbGetBi
+    :: forall v m.
+       (Bi v, MonadDBRead m)
+    => DBTag -> ByteString -> m (Maybe (Word8, v))
```

Comment:
Why does this function return `Maybe (Word8, v)` instead of `v`? The only thing one can do with `Word8` is to check that it equals to `dbSerializeVersion` and it should be done in this function.
I see that it's done in `dbDecode`, but after it's done there is no reason to return this `Word8`. Caller can be sure it is `dbSerializeVersion`.